src/main/java/com/gmail/iikaliada/service/CustomUserDetailsService.java

```diff
@@ -0,0 +1,51 @@
+package com.gmail.iikaliada.service;
+
+import com.gmail.iikaliada.entity.Authority;
+import com.gmail.iikaliada.entity.User;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+
+@Service("customUserDetailsService")
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+
+    private final UserService userService;
+
+    @Override
+    @Transactional(propagation = Propagation.REQUIRED)
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userService.findUserByUserName(username);
+        if (user == null) {
+            throw new UsernameNotFoundException(username);
+        }
+        if (isDateTimeLaterByMinutes(user.getLastTimeFailed(), 5)) {
```

Comment:
5 is the MAGIC NUMBER here. There are two possible approaches:
1. To move to constant
2. To move it to application.yml config. 

The second case is better as you will be able to change value easily without recompilation of source code.